### PR TITLE
daemon: apply Redactor to intent.prompt_preview

### LINKED ISSUE #1012: daemon: prompt_preview is persisted to signed receipts without redaction
`intentFromFrame` (`daemon/internal/pipeline/build.go:955-972`) copies emitter-supplied `f.PromptPreview` into `receipt.Intent` applying **truncation only**. The `Redactor` is never applied — its sole call site in the daemon is `errText` (`build.go:858-859`).

The Redactor's own doc comment (`build.go:214-219`) explains the narrow coverage:

> Redactor is applied to text fields before they are persisted in the receipt body. Today that means `outcome.error` only; input and output are never stored as raw text — only their SHA-256 hashes go into `parameters_hash` / `response_hash`.

That reasoning holds for input/output but does not cover `prompt_preview`, which **is** stored raw. The very next comment (`build.go:221`) contradicts the gap: it describes "the **two** plaintext fields the daemon writes inline: `outcome.error` and [`prompt_preview`]". The code identifies two inline plaintext fields and redacts one.

### Why it matters
A prompt preview is among the likeliest places for a secret to appear — API keys and connection strings get pasted into agent prompts routinely. The daemon carries 30 sensitive-key rules and 13 secret-detection patterns, wired up and operator-configurable (`daemon.go:638`), that never touch this field. mcp-proxy's architecture docs state the daemon owns redaction, so no emitter is expected to sanitise first.

### Current exposure: none known
No in-tree emitter populates `prompt_preview` — `rg` across `hook/`, `mcp-proxy/`, and `integrations/` returns zero hits, and the hook's transcript reader deliberately projects out only `message.model` and `message.usage` (`claude_transcript.go:19-31`). So this is latent today.

It is, however, a documented wire-protocol field, so any third-party emitter that populates it writes unredacted prompt text into a signed, shareable audit record.

### Suggested fix
Apply `p.Redactor` to the preview in `intentFromFrame` — after truncation, matching the ordering rationale already given for `errText` at `build.go:852-856` (redaction can lengthen a string, so capping first can still leave an over-cap value). Update the `Redactor` doc comment so it stops implying `outcome.error` is the only inline plaintext field.

_Surfaced by a code-review pass._

### PR DESCRIPTION
**Pull request by ojongerius**
## What

`intentFromFrame` (`daemon/internal/pipeline/build.go`) now runs `f.PromptPreview` through `p.Redactor` before truncating it, matching the redact-then-truncate ordering already used for `outcome.error` in `buildAndSign`. Also updates the stale `Redactor` doc comment on `Pipeline`, which claimed `outcome.error` was the only field it covered.

## Why

`intentFromFrame` truncated `prompt_preview` but never redacted it, even though the daemon's `Redactor` (30 sensitive-key rules, 13 secret-detection patterns) was already wired up and applied to `outcome.error` at the same call site. A prompt preview is one of the likeliest places for a pasted secret to appear, so this left a documented wire-protocol field free to write unredacted plaintext into signed, shareable receipts.

Redaction runs before truncation for the same reason it does for `errText`: redaction can lengthen a string (a short secret replaced by the longer `[REDACTED]` placeholder), so capping first could leave an over-cap value.

No in-tree emitter currently populates `prompt_preview`, so this closes a latent gap rather than a live leak.

Fixes #1012

## Checklist

- [x] Tests pass for all changed components
- [x] Linter passes (`go vet`, `ruff check`, `biome` as applicable)
- [x] No real keys or secrets in the diff
- [x] Cross-language tests pass (if receipt format, signing, or hashing changed) — N/A, no receipt format change
- [x] AGENTS.md updated (if project structure changed) — N/A, no structure change
- [x] Spec changes have been reviewed by a maintainer (if applicable) — N/A, no spec change

## Security

- [x] This PR touches crypto, auth, or secrets handling
- [x] Primitives and parameters have been reviewed — no crypto/primitive change, redaction logic reused as-is
- [x] All inputs are validated at trust boundaries — `Redactor.Redact` already handles empty/nil safely; `intentFromFrame` guards nil `redactor`
- [x] Edge cases are tested (nil, empty, corrupted, concurrent) — added tests cover redaction-then-truncation ordering and interaction with the existing prompt-preview truncation/cap tests

### CODE REVIEW

File: daemon/internal/pipeline/redact_test.go [hunk 1]
```diff
@@ -395,6 +395,97 @@ func TestPipeline_RedactionAppliedToError(t *testing.T) {
...
+	if !strings.Contains(intent.PromptPreview, "[REDACTED]") {
+		t.Errorf("prompt_preview does not contain [REDACTED]: %q", intent.PromptPreview)
+	}
+}
+
+// TestPipeline_PromptPreviewRedactedBeforeTruncation verifies redaction runs
+// before the rune cap, matching the ordering already used for outcome.error
+// (build.go:852-856): redaction can lengthen a string (a short secret replaced
+// by the longer "[REDACTED]" placeholder), so capping first could leave an
+// over-cap value, and redacting after truncation could leave a secret's tail
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply daemon/internal/pipeline/redact_test.go]: Good catch — fixed in 94b1aad. The cap is now 30 (between the redacted length of 37 and the point where a raw truncation would cut the `ghp_` token short of its `{36,}` minimum), and the test asserts the stored preview contains `[REDACTED]`, contains no raw `ghp_` prefix, and is flagged truncated. Verified it actually discriminates ordering by temporarily reverting the fix and confirming the test fails (raw `ghp_xxx...` leaks) before restoring it.

**Comment by ojongerius** [comment]
Ran a self-review pass on this diff and found two things:

1. **Confirmed**: `Redactor.Redact()`'s JSON-aware pass silently corrupts valid-JSON input (large-integer precision loss via `float64` round-trip, and key reordering from re-marshaling a map). This is pre-existing — already affects `outcome.error` — but this PR increases exposure by routing a second, more free-text-shaped field (`prompt_preview`) through the same path. Deliberately kept out of this PR's scope since fixing it properly touches shared redaction logic used by both fields; filed as **#1022**.
2. Small duplication: the "redact if a Redactor is configured" guard was repeated verbatim at both call sites (`errText` and `prompt_preview`). Extracted `Redactor.RedactIfSet` (nil-receiver method) in bcb764f so the invariant lives in one place.

All tests/vet/build still green after the refactor.
